exercise/exercise_models.py

```diff
@@ -706,6 +706,10 @@ def grade(self, request, submission, no_penalties=False, url_name="exercise"):
         """
         # Get the language from the submission
         language = submission.lang
+        if language is None:
+            language = submission.exercise.course_module.course_instance.language
```

Comment:
As `self` is the exercise, then `self.course_module.course_instance.language` should be enough. However, the language could be `|en|fi|`, thus you need to take the first. As I think this might be relatively common, I think we should add a property to the course instance `default_language`, which would contain the logic for that (so, `self.course_module.course_instance.default_language`).